### PR TITLE
Tracing: Exclude query string for `http.url` tag from `net/http` instrumentation 

### PR DESCRIPTION
**Pull request by TonyCTHsu**
**What does this PR do?**

Exclude query string for `http.url` tag from `net/http` instrumentation

### CODE REVIEW

File: spec/datadog/tracing/contrib/ethon/easy_patch_spec.rb [hunk 1]
```diff
@@ -248,4 +248,15 @@
       expect(span.get_tag('out.host')).to eq('example.com')
     end
   end
+
+  context 'when query string in url' do
+    it 'does not collect string' do
```
Comment (marcotc) [inline spec/datadog/tracing/contrib/ethon/easy_patch_spec.rb:253]:
```suggestion
    it 'does not collect the query string' do
```

File: spec/datadog/tracing/contrib/ethon/easy_patch_spec.rb [hunk 2]
```diff
@@ -248,4 +248,15 @@
...
+      easy = EthonSupport.ethon_easy_new(url: 'http://example.com/sample/path?foo=bar')
+
+      easy.perform
+
+      expect(span.get_tag('http.url')).to eq('/sample/path')
+      expect(span.get_tag('out.host')).to eq('example.com')
```
Comment (marcotc) [inline spec/datadog/tracing/contrib/ethon/easy_patch_spec.rb:259]: I don't think we need to test for hostname here, given this test is asserting on query strings.

**Review by marcotc — APPROVED** [review]
Great work, @TonyCTHsu! 🙇

File: lib/datadog/tracing/contrib/http/instrumentation.rb [hunk 3]
```diff
@@ -79,8 +80,10 @@ def annotate_span_with_request!(span, request, request_options)
 
               span.set_tag(Tracing::Metadata::Ext::TAG_COMPONENT, Ext::TAG_COMPONENT)
               span.set_tag(Tracing::Metadata::Ext::TAG_OPERATION, Ext::TAG_OPERATION_REQUEST)
-
-              span.set_tag(Tracing::Metadata::Ext::HTTP::TAG_URL, request.path)
+              span.set_tag(
+                Tracing::Metadata::Ext::HTTP::TAG_URL,
+                Contrib::Utils::Quantization::HTTP.url(request.path, { query: { exclude: :all } })
```
Comment (delner) [inline lib/datadog/tracing/contrib/http/instrumentation.rb:85]: This seems fine as a default, and you've added tests to make sure there aren't any query strings in other integrations.

Is it worth putting this same quantization line in those integrations as well? For consistency? Or is there a drawback?

**Review by delner — APPROVED** [review]
Perhaps you should address @marcotc 's comment first, and I have one question here, but otherwise this seems acceptable.

File: lib/datadog/tracing/contrib/http/instrumentation.rb (hunk 3 above)
Comment (TonyCTHsu) [reply lib/datadog/tracing/contrib/http/instrumentation.rb:85]: Currently, other http client integration do not offer settings for quantization, each of them leverage their public interface to provide data. 

I believe we should refactor and centraliz the qunatization mechanism when we offer the feature.

**Pull request merged**